.github/workflows/run_test_windows.yml

```diff
@@ -0,0 +1,287 @@
+              with:
+                  directories: ${{ inputs.directories }}
+                  exclude-dirs: ${{ inputs.exclude-dirs }}
+
+    run-test:
+        name: '${{matrix.test.name}} | ${{matrix.test.displayPath}} | ${{matrix.azle_source}} | WSL'
+        needs: get-test-infos
+        runs-on: windows-latest
+        env:
+            ETHEREUM_URL: ${{ secrets.ETHEREUM_URL }}
+            AZLE_IDENTITY_STORAGE_MODE: 'plaintext'
+            AZLE_END_TO_END_TEST_LINK_AZLE: ${{ matrix.azle_source == 'repo' }}
+
+        strategy:
+            fail-fast: false # We want to see which example tests succeed and which ones fail, we don't want one example test to cancel the rest
+            matrix: # spins up one job per combination of test and code source (repo or npm).
+                azle_source:
+                    - ${{ inputs.link-azle == true && 'repo' || 'npm' }}
+                test: ${{ fromJSON(needs.get-test-infos.outputs.test-infos) }}
+        steps:
+            - name: Setup WSL
+              uses: Vampire/setup-wsl@v3
+              with:
+                  distribution: Ubuntu-22.04
+                  update: 'true'
+                  additional-packages: curl build-essential git jq
+                  wsl-conf: |
+                      [boot]
+                      systemd=true
+                  set-as-default: 'true'
+                  wsl-shell-user: 'root'
+
+            - name: Upgrade to WSL 2
+              run: |
+                  echo "Current WSL status before upgrade:"
+                  wsl --status
+
+                  echo "Updating WSL kernel..."
+                  wsl --update --web-download
+
+                  echo "Setting default version to WSL 2..."
+                  wsl --set-default-version 2
+
+                  echo "Converting Ubuntu to WSL 2..."
+                  wsl --set-version Ubuntu 2
+
+                  echo "WSL status after upgrade:"
+                  wsl --status
+                  wsl --list --verbose
+              shell: powershell
+
+            - uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0
+
+            - name: Configure git for Unix line endings
+              run: |
+                  # Configure git to use LF line endings and fix any existing files
+                  git config core.autocrlf false
+                  git config core.eol lf
+                  # Reset and checkout to fix line endings
+                  git rm --cached -r .
+                  git reset --hard
+              shell: wsl-bash {0}
+
+            - uses: ./.github/actions/determine_workflow_context
+              id: set-conditions
+
+            - name: Print environment variables
+              run: |
+                  # Set workflow context variables inline
+                  AZLE_IS_MAIN_BRANCH_PUSH_FROM_FEATURE_MERGE="${{ steps.set-conditions.outputs.is_main_branch_push_from_feature_merge || 'false' }}"
```

Comment:
Why are we setting these environment variables so much differently than in the linux run_test?